extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -406,6 +408,7 @@ export class AmpSidebar extends AMP.BaseElement {
   triggerEvent_(name) {
     const event = createCustomEvent(this.win, `${TAG}.${name}`, {});
     this.action_.trigger(this.element, name, event, ActionTrust.HIGH);
+    this.element.dispatchCustomEvent(name);
```

Comment:
The only issue with using `listenOncePromise` is that we must use `dispatchCustomEvent` in order to have something to listen to. I'm not sure if this is consistent with the original intention of @aghassemi 's PR, but this seems to be the default way that I've seen other components implement this.

Reply:
You can use `dispatchCustomEventForTesting` if just needed for testing.

`dispatchCustomEvent` alone (without `action_.trigger`) won't make the event available for consumption using AMP actions.

`dispatchCustomEvent` is normally used for messaging between components and/or services in cases where we don't want to hold references. The new `signals` framework can actually replace most cases where `dispatchCustomEvent` is used.

Reply:
Great! Thanks!

Reply:
another benefit of `*ForTesting` methods is that they get removed from the production build, reducing code size